type-c-service/src/wrapper/mod.rs

```diff
@@ -404,7 +406,11 @@ where
 
                             // No more awaits, modify state here for drop safety
                             let sw_event = core::mem::replace(
-                                &mut state.port_states_mut()[port_index].sw_status_event,
+                                &mut state
+                                    .port_states_mut()
+                                    .get_mut(port_index)
+                                    .ok_or(Error::Pd(PdError::InvalidPort))?
```

Comment:
An invalid port will mean the `controller.clear_port_events()` already happened but gets dropped. Can the `&mut T` should be fetched before any state-modifying operation happens?